logfire/experimental/read_client.py

```diff
@@ -0,0 +1,338 @@
+from __future__ import annotations
+
+from datetime import datetime
+from types import TracebackType
+from typing import TYPE_CHECKING, Annotated, Any, Generic, Literal, Self, TypedDict, TypeVar
+
+from annotated_types import Gt, Le
```

Comment:
This isn't in the normal dependencies, either users have to install it or it should only go under TYPE_CHECKING if possible.